File: BOJ/10845.rs
use std::collections::VecDeque;
use std::io::*;

fn main() {
    let input = read_to_string(stdin()).unwrap();

    let queries = input
        .lines()
        .skip(1)
        .map(|line| line.split_ascii_whitespace().collect::<Vec<_>>());

    let mut queue = VecDeque::new();

    let mut process_query = |query: Vec<&str>| {
        let read_int = || -> i32 { query[1].parse().unwrap() };

        match query[0] {
            "push" => {
                queue.push_back(read_int());
                None
            }
            "pop" => Some(queue.pop_front().unwrap_or(-1)),
            "size" => Some(queue.len() as i32),
            "empty" => Some(queue.is_empty() as i32),
            "front" => Some(*queue.front().unwrap_or(&-1)),
            "back" => Some(*queue.back().unwrap_or(&-1)),
            _ => None,
        }
    };

    let output = queries
        .filter_map(|query| process_query(query))
        .map(|x| x.to_string())
        .collect::<Vec<_>>()
        .join("\n");

    print!("{output}");
}

File: BOJ/10845.cpp
#include <iostream>
#include <queue>

using namespace std;

queue<int> q;

void push(int num) {
  q.emplace(num);
}

bool empty() {
  return q.empty();
}

int pop() {
  if (empty()) return -1;

  int res = q.front();
  q.pop();
  return res;
}

int size() {
  return (int) q.size();
}

int front() {
  if (empty()) return -1;

  return q.front();
}

int back() {
  if (empty()) return -1;

  return q.back();
}

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(nullptr);

  int n;
  cin >> n;
  while (n--) {
    string cmd;
    cin >> cmd;
    if (cmd == "push") {
      int num;
      cin >> num;
      push(num);
    } else if (cmd == "pop") {
      cout << pop() << '\n';
    } else if (cmd == "size") {
      cout << size() << '\n';
    } else if (cmd == "empty") {
      cout << empty() << '\n';
    } else if (cmd == "front") {
      cout << front() << '\n';
    } else if (cmd == "back") {
      cout << back() << '\n';
    }
  }
  return 0;
}